entities/src/main/java/com/flipkart/varadhi/entities/SubscriptionResource.java

```diff
@@ -0,0 +1,40 @@
+package com.flipkart.varadhi.entities;
+
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.NotNull;
+import lombok.Getter;
+
+@Getter
+@ValidateResource(message = "Invalid Subscription name. Check naming constraints.", max = 64)
+public class SubscriptionResource extends VersionedEntity implements Validatable {
```

Comment:
should we also add an optional param to start the subscription on creation itself?
This feature is mentioned in the consume flow doc.